apps/api/internal/repository/organization_user.go

```diff
@@ -57,3 +59,154 @@ func (r *OrganizationUserRepository) GetOrganizationUser(user_id, organization_i
 
 	return role, nil
 }
+
+func (r *OrganizationUserRepository) UpdateOrganizationUserRole(organizationID, userID, newRole string) error {
+	// First, check if the user is related to the organization
+	checkQuery := `
+		SELECT 1 FROM organization_users
+		WHERE organization_id = $1 AND user_id = $2
+	`
```

Comment:
Do we really need this check? before calling the UpdateOrganizationUserRole we are checking the user role. If the user role is empty this means that the user not exist.